loopy/kernel/__init__.py

```diff
@@ -766,11 +780,10 @@ def iname_tags_of_type(self, iname, tag_type_or_types,
 
     @memoize_method
     def all_inames(self):
-        result = set()
-        for dom in self.domains:
-            result.update(
-                    intern(n) for n in dom.get_var_names(dim_type.set))
-        return frozenset(result)
+        """
+        Returns a :class:`frozenset` of the names of all the inames in the kernel.
+        """
+        return frozenset(iname.name for iname in self.inames)
```

Comment:
I think this should use the domains as a source of truth.

Reply:
Yep, agreed. 8b3131e